src/javascript/app/Stores/client-store.js

```diff
@@ -230,6 +230,11 @@ export default class ClientStore extends BaseStore {
     switchAccount(loginid) {
         this.root_store.ui.removeAllNotifications();
         this.switched = loginid;
+
+        if (location.pathname.includes('contract')) {
+            this.root_store.modules.contract.onUnmountReplay();
+            location.reload();
```

Comment:
is there anything we can do other than reload?

Reply:
since the only way to switch account from `/contract/` is from the 404 I think so?